### PR TITLE
simplify installation documentation

### LINKED ISSUE #2498: [Doc re-organization] - [Linux script] - [Task]
When reading the linux script part I don't really understand the responsibilities of the linux script flags.

Some questions have been added in [link](https://hypothes.is/search?q=tag:%22freq_linux_script%22) are just examples and not exhaustive. **Feel free to add new ones.**

A way to review this section is : how to explain what behind the install flags with a functional way.
Example : Mandatory software as: ta-lib -> the libraries to create trading technical indicators (ta-lib, qtlib)
Like the minimum knowledge that i have to know about what do the script about the feature he provide me (i don't care he create a folder with admin rights).

### PR DESCRIPTION
**Pull request by xmatthias**
## Summary
This should simplify the installation documentation.
Docker should be the first option for new users (i think that's the easiest to get started with).
Also, Exchange keys are NOT part of the installation - users won't need them until the strategy is ready and backtested.

Closes #2498

## Quick changelog

- adjust easy-install script (i'll honestly not put too much work into this, as i see this method disappear hopefully soon)
- Reorder points to have docker installation come first on the documentation page
- api-keys should be part of the configuration - not part of the installation.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -647,11 +652,14 @@ you run it in production mode.
...
 ```
 
-!!! Note
-    If you have an exchange API key yet, [see our tutorial](installation.md#setup-your-exchange-account).
-
 You should also make sure to read the [Exchanges](exchanges.md) section of the documentation to be aware of potential configuration details specific to your exchange.
 
+### Setup your exchange account
+
+You will need to create API Keys (usually you get `key` and `secret`) from the Exchange website and you'll need to insert this into the appropriate fields in the configuration or when asked by the installation script.
```
Comment (hroff-1902) [inline docs/configuration.md]:
```suggestion
You will need to create API Keys (usually you get `key` and `secret`, some exchanges supply it with `password`) from the Exchange website and you'll need to insert this into the appropriate fields in the configuration or when asked by the installation script.
```

File: docs/configuration.md [hunk 2]
```diff
@@ -647,11 +652,14 @@ you run it in production mode.
...
+API Keys are usually only required for real / production trading, but are not required for paper-trading / dry-run.
```
Comment (hroff-1902) [inline docs/configuration.md]:
```suggestion
API Keys are usually only required for live trade (trading for real money, bot running in the "production mode", executing real orders on the exchange) and are not required for the bot running in the dry-run (trade simulation) mode. When you setup the bot in the dry-run mode, you may fill these fields with empty values.
```

File: mkdocs.yml [hunk 3]
```diff
@@ -1,8 +1,8 @@
 site_name: Freqtrade
 nav:
-    - About: index.md
-    - Installation: installation.md
+    - Home: index.md
     - Installation Docker: docker.md
+    - Installation: installation.md
```
Comment (hroff-1902) [inline mkdocs.yml]: I understand your intend (to have docker as the main/easiest option) and I do not mind against this, but you moved the more particular doc page (Installation Docker) to be above the more general one (Installation), I guess this should not be this way. Later, it (Installation page) can be splitted into general and quick-install pages and the order can be
* Installation (general page, prewords/prerequisites, review of available methods and workflow)
* Installation Docker
* Installation Quick-Install script (Linux)
* Manual installation (Linux)
* Installation on Windows
* Installation on RPI
or something like that. Simple moving Installation Docker to be above Installation seems wrong for me.

Comment (xmatthias) [reply mkdocs.yml]: Yes i know - and i did this with full intend.

everything in the installation.md file is "advanced installation". The only section that wasn't was the exchange key generation - which honestly, is wrong to have in an installation documentation in the first place - as it's not needed until you move the bot to production.

If you go to the freqtrade main doc page https://www.freqtrade.io/en/latest/ - and assume you're a new user - the first thing you'll look for is features - then "how to install" - so as user, i pick the uppermost "install" point - assumign that's the easiest (but currently it's not).
Assuming users scroll all the way to the bottom of that page to find the install link is ... optimistic at best... maybe if they're not sure and read it all - but if the user has decided to install - that's probably not what [s]he'll do.

The `docker.md` page on the other hand contains everything you need to get up and running with docker (no need to check `installation.md`), so to me, this should be at the top - as i hope it'll remove some of the install problems ta-lib sometimes causes.

I'm also not keen on splitting the installation documentation too much / further ... having too many options in the left-hand menu might help you find the correct advanced installation method - however, it'll also make it more difficult to find other options once you're installed ... (at which point you probably don't need "installation" anymore).

File: docs/configuration.md [hunk 4]
```diff
@@ -626,6 +626,11 @@ In production mode, the bot will engage your money. Be careful, since a wrong
 strategy can lose all your money. Be aware of what you are doing when
 you run it in production mode.
 
+### Setup your exchange account
```
[the remark this answers is not in the document]
Comment (xmatthias) [reply docs/configuration.md]: arrr yes i messed up and copied it into 2 places, and modified it only in one :D 

regarding the telegram bot - i know it's missing now - but it's not related to setting up exchange keys, nor is it directly related to the installation (where it was before).
So honestly, i do struggle to find an appropriate place for this.

File: mkdocs.yml (hunk 3 above)
Comment (hroff-1902) [reply mkdocs.yml]: > I'm also not keen on splitting the installation documentation too much / further ... having too many options in the left-hand menu might help you find the correct advanced installation method - however, it'll also make it more difficult to find other options once you're installed ... (at which point you probably don't need "installation" anymore).

That's why the installation page should be split (probably as I suggested above) and installation pages moved into a submenu, like we have for Controlling the bot -- they will be hidden when you don't need them anymore. With a small general page for installation methods at the top of the set of the pages.

Okay, let's try to see this page above the general Installation, but this is temporary...

**Pull request merged**